### PR TITLE
Added flag for filtering attributes on product grid

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for a41e883e9513650da0a403e3910e83ca94506b67 (NyanKiyoshi/saleor @ attributes/grid-flag)
Base comparison is c0cd7c427938b8b540259e289ef972580e5788e3.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         43	         43	             21

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         15	         15	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	         15	         15	              6
```

</p>
</details>

File: saleor/graphql/product/types/attributes.py [hunk 1]
```diff
@@ -113,6 +113,10 @@ class Attribute(CountableDjangoObjectType):
         graphene.Boolean(description=AttributeDescriptions.FILTERABLE_IN_DASHBOARD),
         model_field="filterable_in_dashboard",
     )
+    available_in_grid = gql_optimizer.field(
```
Comment (maarcingebala) [inline saleor/graphql/product/types/attributes.py]: Why is this using the optimizer? I guess it's not required for a simple boolean field.

File: saleor/graphql/product/mutations/attributes.py [hunk 2]
```diff
@@ -44,6 +44,9 @@ class AttributeCreateInput(graphene.InputObjectType):
     storefront_search_position = graphene.Int(
         required=False, description=AttributeDescriptions.STOREFRONT_SEARCH_POSITION
     )
+    available_in_grid = graphene.Int(
```
Comment (maarcingebala) [inline saleor/graphql/product/mutations/attributes.py]: Not `graphene.Boolean`?

File: saleor/graphql/product/mutations/attributes.py [hunk 3]
```diff
@@ -75,6 +78,9 @@ class AttributeUpdateInput(graphene.InputObjectType):
...
```
Comment (maarcingebala) [inline saleor/graphql/product/mutations/attributes.py]: Ditto.

**Pull request merged**